pkg/network/namescheme/networknamescheme.go

```diff
@@ -138,6 +138,16 @@ func PodHasOrdinalInterfaceName(podNetworkStatus map[string]networkv1.NetworkSta
 	return false
 }
 
+// PodHasOrdinalInterfaceName2 checks if the given pod network status has at least one pod interface with ordinal name
+func PodHasOrdinalInterfaceName2(podNetworkStatuses []networkv1.NetworkStatus) bool {
+	for _, networkStatus := range podNetworkStatuses {
```

Comment:
We do not index for performance, we index because it hurts our thinking to see searches being repeated again and again. Indexing performs the mapping at one step and then the usages can reuse it.

I guess we got used to this pattern so we keep using it.

Copying if the range for loop is not always expensive, it depends on the amount of data. I think there is a linter that raises an error when we copy large structures.
In the range loop it is even cheaper, as the variable is re-used and not re-allocated all the time.

Bottom line, we only use indexing if we need to access the original data and mutate it (either in the loop or when we return a pointer to an entry).